root/static/scripts/edit/URLCleanup.js

```diff
@@ -2106,7 +2107,10 @@ const CLEANUPS = {
       const m = /^https:\/\/twitter\.com\/[^\/?#]+(\/status\/\d+)?$/.exec(url);
       if (m) {
         const isATweet = !!m[1];
-        return isATweet === (id === LINK_TYPES.streamingmusic.recording);
+        if (_.includes(LINK_TYPES.streamingmusic, id)) {
+          return isATweet && (id === LINK_TYPES.streamingmusic.recording);
+        }
+        return !isATweet;
```

Comment:
Can you please explain what this one does to me? :)

Reply:
There was no validation for Twitter accounts in the previous version.

This one auto-selects `streamingmusic` for any Twitter link added to recording/release, validates tweets only for recording, and invalidates any Twitter link for release. It keeps auto-selecting `socialnetwork` for other entity types (in `socialnetwork`) and validates Twitter accounts only for these.

* v1 has :-1: two CLEANUPS entries with :-1: duplicated code (`match` and `cleanup`), :+1: simple `type`s and :+1: only one simple `validate` for tweets:
https://github.com/metabrainz/musicbrainz-server/blob/2c1b2a250568744e185ee0a557b7346ec054f190/root/static/scripts/edit/URLCleanup.js#L2088-L2124

* v2 has :+1: only one CLEANUPS entry with :+1: no duplicated code, :grey_question: aggregated `type` and :grey_question: slightly more complicated `validate` for both tweets and accounts:
  https://github.com/metabrainz/musicbrainz-server/blob/36e7b43b58fa8dc4a460a225f0a5b29b2f716138/root/static/scripts/edit/URLCleanup.js#L2088-L2117

----

P.S. I can actually simplify type aggregation by removing
```javascript
      {recording: LINK_TYPES.streamingmusic.recording},
```
from `type` as `socialnetwork` has no `recording` entry.

Reply:
I meant from the previous version in the previous commit where it only has the isATweet check, but I understand that's to block them completely from release? I guess that's fine then :)

Reply:
OK, the last commit adds relationship type auto-select for Twitter URLs on releases and invalidates such link. Linking to Twitter on releases used to be unchecked and thus free for linking with any unrestricted relationship type.

Both versions do prevent linking tweets to releases, differences are mostly about coding style.